# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /home/henrik/NetBeansProjects/hudson/plugins/downstream-buildview/src/main/java/org/jvnet/hudson/plugins
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: DownstreamBuildViewUpdateListener.java
--- DownstreamBuildViewUpdateListener.java Base (BASE)
+++ DownstreamBuildViewUpdateListener.java Locally Modified (Based On LOCAL)
@@ -70,10 +70,15 @@
     public void onStarted(AbstractBuild r,TaskListener listener) {
     	//build = r;
     	CauseAction ca = r.getAction(CauseAction.class);
-    	UpstreamCause upcause =null;
+
+
+        if (ca == null || ca.getCauses() ==null) {
+            return;
+        }
+
     	for (Cause c : ca.getCauses()){
     		if( c instanceof UpstreamCause){
-    			upcause = (UpstreamCause)c;
+    			UpstreamCause upcause = (UpstreamCause)c;
     			String upProjectName = upcause.getUpstreamProject();
     			int buildNumber = upcause.getUpstreamBuild();
     			AbstractProject project = Hudson.getInstance().getItemByFullName(upProjectName, AbstractProject.class);
